### PR TITLE
fix(calling): failover immediately after 4 keepalive failures for cc flow

### PR DESCRIPTION
**Pull request by rarajes2**
# COMPLETES #< [SPARK-569580](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-569580) >

## This pull request addresses

Failover to secondary mobius server immediately after 4 keep alive failure for contact center flow.

Here we have introduced a new variable/flag to check we need to failover immediately or not. That flag gets set only when the flow is contact centre and keepalive is failing. It gets reset back when the immediate failover logic runs.

## by making the following changes

Able to failover to backup mobius server immediately after 4 keepalive failure.

**Note**: Please check the [HAR](https://jira-eng-gpk2.cisco.com/jira/secure/attachment/1324962/failover-after-keepalive-public-repo.har) file attached on the JIRA.

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by Shreyas281299 — APPROVED** [review]
LGTM

File: packages/calling/src/CallingClient/registration/register.test.ts [hunk 1]
```diff
@@ -732,7 +732,7 @@ describe('Registration Tests', () => {
       expect(reg.getStatus()).toBe(RegistrationStatus.INACTIVE);
       expect(reg.keepaliveTimer).not.toBe(timer);
 
-      webex.request.mockRejectedValueOnce(failurePayload).mockResolvedValue(successPayload);
+      webex.request.mockResolvedValue(successPayload);
```
Comment (sreenara) [inline packages/calling/src/CallingClient/registration/register.test.ts:735]: How does this change ensure we are failing over immediately? I'm not clear about how this statement change covers the new flow.

**Review by sreenara — CHANGES_REQUESTED** [review]
Please add some more details to the description on why we are using a new variable to control this immediate failover. That'll help others to return to this PR in the future and understand the change better.

File: packages/calling/src/CallingClient/registration/register.test.ts (hunk 1 above)
Comment (rarajes2) [reply packages/calling/src/CallingClient/registration/register.test.ts:735]: The failure payload was for the primary server try but now we don't try primary server for the cc flow when it comes from the keepalive failure, here we show the success for the secondary server immediately.

**Review by sreenara — COMMENTED** [review]
> Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.

@rarajes2 in the HAR file 'failover-after-keepalive-public-repo.har', it looks like the SDK tries to register with the primary twice before failing over to secondary. Why is that happening?

**Comment by rarajes2** [comment]
> > Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.
> 
> @rarajes2 in the HAR file 'failover-after-keepalive-public-repo.har', it looks like the SDK tries to register with the primary twice before failing over to secondary. Why is that happening?

@sreenara Fixed that. It was happening because we have logic to reregister (a function) after keepalive failures which first tries to attempt primary server then starts the failover. Updated the logic in both the PR and also updated JIRA with latest HAR files.

**Review by sreenara — APPROVED** [review]

**Pull request merged**